File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IPhotoAllCommentView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.Comment;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IPhotoAllCommentView extends IAccountDependencyView, IMvpView, IErrorView, IAttachmentsPlacesView {
    void displayData(List<Comment> comments);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);

    void dismissDeepLookingCommentProgress();

    void displayDeepLookingCommentProgress();

    void moveFocusTo(int index, boolean smooth);

    void notifyDataAddedToTop(int count);

    void notifyItemChanged(int index);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IPhotoAllCommentView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Comment
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IPhotoAllCommentView : IAccountDependencyView, IMvpView, IErrorView, IToastView,
    IAttachmentsPlacesView {
    fun displayData(comments: MutableList<Comment>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
    fun dismissDeepLookingCommentProgress()
    fun displayDeepLookingCommentProgress()
    fun moveFocusTo(index: Int, smooth: Boolean)
    fun notifyDataAddedToTop(count: Int)
    fun notifyItemChanged(index: Int)
}